tapir/coop/forms.py

```diff
@@ -247,3 +251,110 @@ class Meta:
         }
 
     share_owner = ShareOwnerChoiceField()
+
+
+class MembershipResignationForm(forms.ModelForm):
+    already_resigned = MembershipResignation.objects.values("share_owner")
+    in_three_years = _("Coop buys back share(s)")
+    cancellation_reason = forms.CharField(
+        max_length=1000,
+        widget=forms.Textarea(
+            attrs={"rows": 2, "placeholder": _("Please not more than 1000 characters.")}
+        ),
+    )
+    coop_buys_shares_back = forms.BooleanField(label=in_three_years, required=False)
+    share_owner = ShareOwnerChoiceField()
+    willing_to_gift_shares_to_coop = forms.BooleanField(
+        label="Willing to gift shares to coop", required=False
+    )
+    transfering_shares_to = ShareOwnerChoiceField(
+        required=False, label=_("Transfering share(s) to")
+    )
+
+    class Meta:
+        model = MembershipResignation
+        fields = [
+            "share_owner",
+            "cancellation_reason",
+            "cancellation_date",
+            "coop_buys_shares_back",
+            "willing_to_gift_shares_to_coop",
+            "transfering_shares_to",
+            "paid_out",
+        ]
+        widgets = {"cancellation_date": DateInputTapir()}
+
+    def clean(self):
+        cleaned_data = super().clean()
+        share_owner = cleaned_data.get("share_owner")
+        coop_buys_shares_back = cleaned_data.get("coop_buys_shares_back")
+        willing_to_gift_shares_to_coop = cleaned_data.get(
+            "willing_to_gift_shares_to_coop"
+        )
+        transfering_shares_to = cleaned_data.get("transfering_shares_to")
+        paid_out = cleaned_data.get("paid_out")
+        cancellation_date = cleaned_data.get("cancellation_date")
+        errmsg = _("Please take only one choice.")
```

Comment:
I would prefer to name that `error_message`, or even `only_one_choice_allowed_error_message`
We should avoid shortened variable names, they are never as clear as full names. With the IDE's auto-completion it shouldn't be longer to type anyway.